src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/SpanNDDebugView.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Numerics.Tensors
+{
+    internal sealed class SpanNDDebugView<T>
+    {
+        private readonly T[] _array;
+
+        public SpanNDDebugView(SpanND<T> span)
+        {
+            _array = span.ToArray();
```

Comment:
I think we need to be careful/consideration here. Unlike most collection types, which tend to be fairly small, tensors are expected to be "large" (megabytes) in many cases.

I don't think we want to risk stalling the debugger because we have to allocate, copy, and then serialize gigabytes of data across the process boundary.

Accordingly, I think we may want to determine if there is a better way to expose the debug view that facilities looking at small 2-3 dimensioned data and which can give users the ability to still view pertinent information for larger tensors.

Reply:
That all makes sense. Is that something we want to decide before the initial preview? Should I just remove it for now? Or should we sit down and figure it out?